dwave/cloud/config.py

```diff
@@ -255,94 +408,92 @@ def get_default_config():
+    environment variable, does not exist or is unreadable, loading fails with
+    :exc:`~dwave.cloud.exceptions.ConfigFileReadError`. Loading fails
+    with :exc:`~dwave.cloud.exceptions.ConfigFileParseError` if the file is
+    readable but invalid as a configuration file.
+
+    Similarly, if a profile explicitly specified, via an argument or
+    environment variable, is not present in the loaded configuration, loading fails
+    with :exc:`ValueError`. Explicit profile selection also fails if the configuration
+    file is not explicitly specified, detected on the system, or defined via
+    an environment variable.
 
     Environment variables:
 
         ``DWAVE_CONFIG_FILE``:
-            Config file path used if ``config_file`` not specified.
+            Configuration file path used if no configuration file is specified.
 
         ``DWAVE_PROFILE``:
-            Name of config profile (section) to use if ``profile`` not specified.
+            Name of profile (section) to use if no profile is specified.
 
         ``DWAVE_API_CLIENT``:
-            API client class used (can be: ``qpu`` or ``sw``). Overrides values
-            from config file, but is overridden with ``client``.
+            API client class used if no client is specified. Supported values are
+            ``qpu`` or ``sw``.
 
         ``DWAVE_API_ENDPOINT``:
-            API endpoint URL to use instead of the URL given in config file,
-            if ``endpoint`` not given.
+            API endpoint URL used if no endpoint is specified.
 
         ``DWAVE_API_TOKEN``:
-            API authorization token. Overrides values from config file, but is
-            overridden with ``token``.
+            API authorization token used if no token is specified.
 
         ``DWAVE_API_SOLVER``:
-            Default solver. Overrides values from config file, but is overridden
-            with ``solver``.
+            Default solver used if no solver is specified.
 
         ``DWAVE_API_PROXY``:
-            URL for proxy to use in connections to D-Wave API. Overrides values
-            from config file, but is overridden with ``proxy``.
+            URL for proxy connections to D-Wave API used if no proxy is specified.
 
     Args:
 
         config_file (str/None/False/True, default=None):
-            Path to config file.
+            Path to configuration file.
 
-            If undefined (set to ``None``), the name of the config file is
-            taken from ``DWAVE_CONFIG_FILE`` environment variable.
-            If that env var is undefined or empty, the location of configuration
-            files is auto-detected, as described in :func:`load_config_from_files`.
+            If ``None``, the value is taken from ``DWAVE_CONFIG_FILE`` environment
+            variable if defined. If the environment variable is undefined or empty,
+            auto-detection searches for existing configuration files in the standard
+            directories set by homebase_.
```

Comment:
It would be more technically accurate (here and elsewhere) to refer to `get_configfile_paths` instead of `homebase`, because:
- the priority (loading order) and detection of config files (system/user) is not defined in `homebase`, but in `get_configfile_paths`
- we added local file (`./dwave.conf`) to system/user dirs returned by `homebase`.